src/server/src/main/java/com/spotify/reaper/resources/view/NodesStatus.java

```diff
@@ -66,75 +86,114 @@ private GossipInfo parseEndpointStatesString(String sourceNode, String allEndpoi
+          severity = Optional.fromNullable(Double.parseDouble(matcher.group(3)));
+          break;
+        }
       }
-      
-      matcher = endpointLoadPattern.matcher(endpointString);
-      if(matcher.find()) {
-        load = Optional.fromNullable(new BigDecimal(matcher.group(3)).doubleValue());
-        totalLoad+=load.or(0.0);
+
+      for (Pattern endpointLoadPattern : endpointLoadPatterns) {
+        matcher = endpointLoadPattern.matcher(endpointString);
+        if (matcher.find()) {
+          load = Optional.fromNullable(new BigDecimal(matcher.group(3)).doubleValue());
+          totalLoad += load.or(0.0);
+          break;
+        }
       }
-      
-      matcher = endpointReleasePattern.matcher(endpointString);
-      if(matcher.find()) {
-        releaseVersion = Optional.fromNullable(matcher.group(3));
+
+      for (Pattern endpointReleasePattern : endpointReleasePatterns) {
+        matcher = endpointReleasePattern.matcher(endpointString);
+        if (matcher.find()) {
+          releaseVersion = Optional.fromNullable(matcher.group(3));
+          break;
+        }
       }
-      
-      matcher = endpointHostIdPattern.matcher(endpointString);
-      if(matcher.find()) {
-        hostId = Optional.fromNullable(matcher.group(3));
+
+      for (Pattern endpointHostIdPattern : endpointHostIdPatterns) {
+        matcher = endpointHostIdPattern.matcher(endpointString);
+        if (matcher.find()) {
+          hostId = Optional.fromNullable(matcher.group(3));
+          break;
+        }
       }
-      
-      matcher = endpointTokensPattern.matcher(endpointString);
-      if(matcher.find()) {
-        tokens = Optional.fromNullable(matcher.group(2));
+
+      for (Pattern endpointTokensPattern : endpointTokensPatterns) {
+        matcher = endpointTokensPattern.matcher(endpointString);
+        if (matcher.find()) {
+          tokens = Optional.fromNullable(matcher.group(2));
+          break;
+        }
       }
-      
-      
-      
+
+
+
       EndpointState endpointState = new EndpointState(endpoint.or(NOT_AVAILABLE), hostId.or(NOT_AVAILABLE), dc.or(NOT_AVAILABLE), rack.or(NOT_AVAILABLE), status.or(NOT_AVAILABLE), severity.or(0.0),
           releaseVersion.or(NOT_AVAILABLE), tokens.or(NOT_AVAILABLE), load.or(0.0));
-      
+
       endpoints.add(endpointState);
     }
-    
+
     Map<String, Map<String, List<EndpointState>>> endpointsByDcAndRack = Maps.newHashMap();
     Map<String, List<EndpointState>> endpointsByDc = endpoints.stream().collect(Collectors.groupingBy(EndpointState::getDc, Collectors.toList()));
-    
+
     for(String dc:endpointsByDc.keySet()) {
       Map<String, List<EndpointState>> endpointsByRack = endpointsByDc.get(dc).stream().collect(Collectors.groupingBy(EndpointState::getRack, Collectors.toList()));
       endpointsByDcAndRack.put(dc, endpointsByRack);
     }
-    
+
     return new GossipInfo(sourceNode ,endpointsByDcAndRack, totalLoad);
   }
-  
-  
+
+  private void initPatterns() {
```

Comment:
this method should be static, and called from a static constructor instead of from `parseEndpointStatesString(..)`

(the lists are going to grow with duplicates and be a computational overhead if multiple NodeStatus instances are instantiated.)